google/cloud/aiplatform/initializer.py

```diff
@@ -88,8 +90,14 @@ def init(
             utils.validate_region(location)
             self._location = location
         if experiment:
-            logging.warning("Experiments currently not supported.")
-            self._experiment = experiment
+            metadata.metadata_service.set_experiment(experiment)
+        if run:
+            if not experiment:
+                raise ValueError(
+                    "No experiment set. Provide an experiment for this run, e.g., aiplatform.init("
+                    "experiment='my-experiment')."
+                )
+            metadata.metadata_service.set_run(run)
```

Comment:
I could be wrong on this as I am not very family how python would handle calls to class methods w/o a created class, but wouldn't this call create a new metadata_service class every time we call the class method and the experiment information is not stored? I thought we would need to add metadata_service(or maybe call it metadata_config) class as a parameter in the global config class so that the metadata status would be stored globally and would be used for check for "log_param/metric" calls. Like what I was trying to do in https://github.com/jialuzh/python-aiplatform/blob/dev-test/google/cloud/aiplatform/initializer.py#L47.

Reply:
Think of the metadata_service as your Config. 
It's a singleton, so is created only once, in the last line of metadata.py. We use that object all over. Similar to what the initializer does